src/Http/Controllers/Controller.php

```diff
@@ -116,32 +116,44 @@ public function insertUpdateData($request, $slug, $rows, $data)
         return $data;
     }
 
-    public function validateBread($request, $data)
+    public function validateBread($request, $data, $action = null)
     {
         $rules = [];
         $messages = [];
+        $customAttributes = [];
 
         foreach ($data as $row) {
+            if ($row->display_name) $customAttributes[$row->field] = $row->display_name;
+
             $options = json_decode($row->details);
 
             if (isset($options->validation)) {
-                if (isset($options->validation->rule)) {
-                    if (!is_array($options->validation->rule)) {
-                        $rules[$row->display_name] = explode('|', $options->validation->rule);
+                $validationOptions = $options->validation;
+
+                // if store or update-specific validation is defined, use that instead
```

Comment:
On the docs there are no mention of different rules per action (update or store). I think if this PR tries to fix an issue, it should do exactly that. A feature should be done in a different commit.